src/builder/create_interaction_response_followup.rs

```diff
@@ -178,7 +186,9 @@ impl Builder for CreateInteractionResponseFollowup {
         ctx: Self::Context<'_>,
     ) -> Result<Self::Built> {
         self.check_length()?;
+
         let files = std::mem::take(&mut self.files);
+        self.attachments.extend(MessageAttachment::from_files(&files));
```

Comment:
20min later, I now understand this line of code (and the long time I needed to understand this is part of my motivation to rename `attachments` + `files` to `attachments_metadata` + `attachments`).

Anyways, I think `self.attachments = ` would be better than `.extend()` because it makes it even clearer that `attachments_metadata` (new naming) _does not_ have any items so far, and will only be filled by this line of code alone

Reply:
Alright. 

> Anyways, I think self.attachments = would be better than .extend() because it makes it even clearer that attachments_metadata (new naming) does not have any items so far, and will only be filled by this line of code alone

This only applies to message creation though. `attachments_metadata` can have entries when editing messages.

Reply:
That is a good point 😵‍💫 This all is confusing.. Maybe there's a way to centralize all this attachment complexity somewhere instead of needing it spread out across all builders.. that'd be cool. But not sure if possible